test/chain/test_node_container.go

```diff
@@ -0,0 +1,169 @@
+package chain
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"path/filepath"
+	"strings"
+
+	"github.com/ory/dockertest"
+	"github.com/ory/dockertest/docker"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	NODE_LABEL_KEY = "ibc-test"
+)
+
+type TestNodeContainer struct {
+	N         *TestNode
+	Config    *ContainerConfig
+	Container *docker.Container
+}
+
+func NewTestNodeContainer(
+	n *TestNode,
+	config *ContainerConfig,
+) *TestNodeContainer {
+	nc := &TestNodeContainer{
+		N:         n,
+		Config:    config,
+		Container: nil,
+	}
+	return nc
+}
+
+// HomeDir returns the home path in the container.
+func (nc *TestNodeContainer) HomeDir() string {
+	return filepath.Join("/home", nc.Config.Bin)
+}
+
+// InitHomeFolder initializes a home folder for the given node.
+func (nc *TestNodeContainer) InitHomeFolder(ctx context.Context) {
+	command := []string{nc.Config.Bin, "init", nc.N.Name(),
+		"--chain-id", nc.N.C.ChainID,
+		"--home", nc.HomeDir(),
+	}
+	nc.RunAndWait(ctx, command, "")
+}
+
+// CreateKey creates a key in the keyring backend test for the given node.
+func (nc *TestNodeContainer) CreateKey(ctx context.Context, name string) {
+	command := []string{nc.Config.Bin, "keys", "add", name,
```

Comment:
*nit* putting the `nc.Config.Bin` on the next line feels like something that go people do more frequently